--- a/components/search_engines/template_url_prepopulate_data.cc
+++ b/components/search_engines/template_url_prepopulate_data.cc
@@ -29,6 +29,7 @@
 
 // Default (for countries with no better engine set)
 const PrepopulatedEngine* const engines_default[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -39,6 +40,7 @@
 // clang-format off
 // United Arab Emirates
 const PrepopulatedEngine* const engines_AE[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -48,6 +50,7 @@
 
 // Albania
 const PrepopulatedEngine* const engines_AL[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -57,6 +60,7 @@
 
 // Argentina
 const PrepopulatedEngine* const engines_AR[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_ar,
@@ -66,6 +70,7 @@
 
 // Austria
 const PrepopulatedEngine* const engines_AT[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_at,
@@ -75,6 +80,7 @@
 
 // Australia
 const PrepopulatedEngine* const engines_AU[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_au,
@@ -84,6 +90,7 @@
 
 // Bosnia and Herzegovina
 const PrepopulatedEngine* const engines_BA[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -93,6 +100,7 @@
 
 // Belgium
 const PrepopulatedEngine* const engines_BE[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -102,6 +110,7 @@
 
 // Bulgaria
 const PrepopulatedEngine* const engines_BG[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -111,6 +120,7 @@
 
 // Bahrain
 const PrepopulatedEngine* const engines_BH[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -120,6 +130,7 @@
 
 // Burundi
 const PrepopulatedEngine* const engines_BI[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -129,6 +140,7 @@
 
 // Brunei
 const PrepopulatedEngine* const engines_BN[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -138,6 +150,7 @@
 
 // Bolivia
 const PrepopulatedEngine* const engines_BO[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -147,6 +160,7 @@
 
 // Brazil
 const PrepopulatedEngine* const engines_BR[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_br,
@@ -156,6 +170,7 @@
 
 // Belarus
 const PrepopulatedEngine* const engines_BY[] = {
+    &baraqah,
     &google,
     &yandex_by,
     &mail_ru,
@@ -165,6 +180,7 @@
 
 // Belize
 const PrepopulatedEngine* const engines_BZ[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -174,6 +190,7 @@
 
 // Canada
 const PrepopulatedEngine* const engines_CA[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_ca,
@@ -183,6 +200,7 @@
 
 // Switzerland
 const PrepopulatedEngine* const engines_CH[] = {
+    &baraqah,
     &google,
     &bing,
     &duckduckgo,
@@ -192,6 +210,7 @@
 
 // Chile
 const PrepopulatedEngine* const engines_CL[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_cl,
@@ -201,6 +220,7 @@
 
 // China
 const PrepopulatedEngine* const engines_CN[] = {
+    &baraqah,
     &baidu,
     &bing,
     &sogou,
@@ -210,6 +230,7 @@
 
 // Colombia
 const PrepopulatedEngine* const engines_CO[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_co,
@@ -219,6 +240,7 @@
 
 // Costa Rica
 const PrepopulatedEngine* const engines_CR[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -228,6 +250,7 @@
 
 // Czech Republic
 const PrepopulatedEngine* const engines_CZ[] = {
+    &baraqah,
     &google,
     &seznam_cz,
     &bing,
@@ -237,6 +260,7 @@
 
 // Germany
 const PrepopulatedEngine* const engines_DE[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_de,
@@ -246,6 +270,7 @@
 
 // Denmark
 const PrepopulatedEngine* const engines_DK[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_dk,
@@ -255,6 +280,7 @@
 
 // Dominican Republic
 const PrepopulatedEngine* const engines_DO[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -264,6 +290,7 @@
 
 // Algeria
 const PrepopulatedEngine* const engines_DZ[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_fr,
@@ -273,6 +300,7 @@
 
 // Ecuador
 const PrepopulatedEngine* const engines_EC[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_es,
@@ -282,6 +310,7 @@
 
 // Estonia
 const PrepopulatedEngine* const engines_EE[] = {
+    &baraqah,
     &google,
     &bing,
     &yandex_ru,
@@ -291,6 +320,7 @@
 
 // Egypt
 const PrepopulatedEngine* const engines_EG[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -300,6 +330,7 @@
 
 // Spain
 const PrepopulatedEngine* const engines_ES[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_es,
@@ -309,6 +340,7 @@
 
 // Finland
 const PrepopulatedEngine* const engines_FI[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_fi,
@@ -318,6 +350,7 @@
 
 // Faroe Islands
 const PrepopulatedEngine* const engines_FO[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_uk,
@@ -327,6 +360,7 @@
 
 // France
 const PrepopulatedEngine* const engines_FR[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_fr,
@@ -336,6 +370,7 @@
 
 // United Kingdom
 const PrepopulatedEngine* const engines_GB[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_uk,
@@ -345,6 +380,7 @@
 
 // Greece
 const PrepopulatedEngine* const engines_GR[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -354,6 +390,7 @@
 
 // Guatemala
 const PrepopulatedEngine* const engines_GT[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -363,6 +400,7 @@
 
 // Hong Kong
 const PrepopulatedEngine* const engines_HK[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_hk,
@@ -372,6 +410,7 @@
 
 // Honduras
 const PrepopulatedEngine* const engines_HN[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -381,6 +420,7 @@
 
 // Croatia
 const PrepopulatedEngine* const engines_HR[] = {
+    &baraqah,
     &google,
     &bing,
     &duckduckgo,
@@ -390,6 +430,7 @@
 
 // Hungary
 const PrepopulatedEngine* const engines_HU[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -399,6 +440,7 @@
 
 // Indonesia
 const PrepopulatedEngine* const engines_ID[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_id,
@@ -408,6 +450,7 @@
 
 // Ireland
 const PrepopulatedEngine* const engines_IE[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_uk,
@@ -417,6 +460,7 @@
 
 // Israel
 const PrepopulatedEngine* const engines_IL[] = {
+    &baraqah,
     &google,
     &bing,
     &yandex_ru,
@@ -426,6 +470,7 @@
 
 // India
 const PrepopulatedEngine* const engines_IN[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_in,
@@ -435,6 +480,7 @@
 
 // Iraq
 const PrepopulatedEngine* const engines_IQ[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -444,6 +490,7 @@
 
 // Iran
 const PrepopulatedEngine* const engines_IR[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -453,6 +500,7 @@
 
 // Iceland
 const PrepopulatedEngine* const engines_IS[] = {
+    &baraqah,
     &google,
     &bing,
     &duckduckgo,
@@ -462,6 +510,7 @@
 
 // Italy
 const PrepopulatedEngine* const engines_IT[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -471,6 +520,7 @@
 
 // Jamaica
 const PrepopulatedEngine* const engines_JM[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -480,6 +530,7 @@
 
 // Jordan
 const PrepopulatedEngine* const engines_JO[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -489,6 +540,7 @@
 
 // Japan
 const PrepopulatedEngine* const engines_JP[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_jp,
@@ -498,6 +550,7 @@
 
 // Kenya
 const PrepopulatedEngine* const engines_KE[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -507,6 +560,7 @@
 
 // South Korea
 const PrepopulatedEngine* const engines_KR[] = {
+    &baraqah,
     &google,
     &naver,
     &bing,
@@ -516,6 +570,7 @@
 
 // Kuwait
 const PrepopulatedEngine* const engines_KW[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -525,6 +580,7 @@
 
 // Kazakhstan
 const PrepopulatedEngine* const engines_KZ[] = {
+    &baraqah,
     &google,
     &yandex_kz,
     &mail_ru,
@@ -534,6 +590,7 @@
 
 // Lebanon
 const PrepopulatedEngine* const engines_LB[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -543,6 +600,7 @@
 
 // Liechtenstein
 const PrepopulatedEngine* const engines_LI[] = {
+    &baraqah,
     &google,
     &bing,
     &duckduckgo,
@@ -552,6 +610,7 @@
 
 // Lithuania
 const PrepopulatedEngine* const engines_LT[] = {
+    &baraqah,
     &google,
     &bing,
     &yandex_ru,
@@ -561,6 +620,7 @@
 
 // Luxembourg
 const PrepopulatedEngine* const engines_LU[] = {
+    &baraqah,
     &google,
     &bing,
     &duckduckgo,
@@ -570,6 +630,7 @@
 
 // Latvia
 const PrepopulatedEngine* const engines_LV[] = {
+    &baraqah,
     &google,
     &bing,
     &yandex_ru,
@@ -579,6 +640,7 @@
 
 // Libya
 const PrepopulatedEngine* const engines_LY[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -588,6 +650,7 @@
 
 // Morocco
 const PrepopulatedEngine* const engines_MA[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_fr,
@@ -597,6 +660,7 @@
 
 // Monaco
 const PrepopulatedEngine* const engines_MC[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_fr,
@@ -606,6 +670,7 @@
 
 // Moldova
 const PrepopulatedEngine* const engines_MD[] = {
+    &baraqah,
     &google,
     &yandex_ru,
     &bing,
@@ -615,6 +680,7 @@
 
 // Montenegro
 const PrepopulatedEngine* const engines_ME[] = {
+    &baraqah,
     &google,
     &bing,
     &yandex_ru,
@@ -624,6 +690,7 @@
 
 // Macedonia
 const PrepopulatedEngine* const engines_MK[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -633,6 +700,7 @@
 
 // Mexico
 const PrepopulatedEngine* const engines_MX[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_mx,
@@ -642,6 +710,7 @@
 
 // Malaysia
 const PrepopulatedEngine* const engines_MY[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_my,
@@ -651,6 +720,7 @@
 
 // Nicaragua
 const PrepopulatedEngine* const engines_NI[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -660,6 +730,7 @@
 
 // Netherlands
 const PrepopulatedEngine* const engines_NL[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_nl,
@@ -669,6 +740,7 @@
 
 // Norway
 const PrepopulatedEngine* const engines_NO[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -678,6 +750,7 @@
 
 // New Zealand
 const PrepopulatedEngine* const engines_NZ[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_nz,
@@ -687,6 +760,7 @@
 
 // Oman
 const PrepopulatedEngine* const engines_OM[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -696,6 +770,7 @@
 
 // Panama
 const PrepopulatedEngine* const engines_PA[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_es,
@@ -705,6 +780,7 @@
 
 // Peru
 const PrepopulatedEngine* const engines_PE[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_pe,
@@ -714,6 +790,7 @@
 
 // Philippines
 const PrepopulatedEngine* const engines_PH[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_ph,
@@ -723,6 +800,7 @@
 
 // Pakistan
 const PrepopulatedEngine* const engines_PK[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -732,6 +810,7 @@
 
 // Poland
 const PrepopulatedEngine* const engines_PL[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -741,6 +820,7 @@
 
 // Puerto Rico
 const PrepopulatedEngine* const engines_PR[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -750,6 +830,7 @@
 
 // Portugal
 const PrepopulatedEngine* const engines_PT[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -759,6 +840,7 @@
 
 // Paraguay
 const PrepopulatedEngine* const engines_PY[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_es,
@@ -768,6 +850,7 @@
 
 // Qatar
 const PrepopulatedEngine* const engines_QA[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -777,6 +860,7 @@
 
 // Romania
 const PrepopulatedEngine* const engines_RO[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -786,6 +870,7 @@
 
 // Serbia
 const PrepopulatedEngine* const engines_RS[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -795,6 +880,7 @@
 
 // Russia
 const PrepopulatedEngine* const engines_RU[] = {
+    &baraqah,
     &yandex_ru,
     &google,
     &duckduckgo,
@@ -804,6 +890,7 @@
 
 // Rwanda
 const PrepopulatedEngine* const engines_RW[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -813,6 +900,7 @@
 
 // Saudi Arabia
 const PrepopulatedEngine* const engines_SA[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -822,6 +910,7 @@
 
 // Sweden
 const PrepopulatedEngine* const engines_SE[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_se,
@@ -831,6 +920,7 @@
 
 // Singapore
 const PrepopulatedEngine* const engines_SG[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_sg,
@@ -840,6 +930,7 @@
 
 // Slovenia
 const PrepopulatedEngine* const engines_SI[] = {
+    &baraqah,
     &google,
     &bing,
     &duckduckgo,
@@ -849,6 +940,7 @@
 
 // Slovakia
 const PrepopulatedEngine* const engines_SK[] = {
+    &baraqah,
     &google,
     &bing,
     &duckduckgo,
@@ -858,6 +950,7 @@
 
 // El Salvador
 const PrepopulatedEngine* const engines_SV[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_es,
@@ -867,6 +960,7 @@
 
 // Syria
 const PrepopulatedEngine* const engines_SY[] = {
+    &baraqah,
     &google,
     &bing,
     &yandex_ru,
@@ -876,6 +970,7 @@
 
 // Thailand
 const PrepopulatedEngine* const engines_TH[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_th,
@@ -885,6 +980,7 @@
 
 // Tunisia
 const PrepopulatedEngine* const engines_TN[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_fr,
@@ -894,6 +990,7 @@
 
 // Turkey
 const PrepopulatedEngine* const engines_TR[] = {
+    &baraqah,
     &google,
     &yandex_tr,
     &yahoo_tr,
@@ -903,6 +1000,7 @@
 
 // Trinidad and Tobago
 const PrepopulatedEngine* const engines_TT[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -912,6 +1010,7 @@
 
 // Taiwan
 const PrepopulatedEngine* const engines_TW[] = {
+    &baraqah,
     &google,
     &yahoo_tw,
     &bing,
@@ -921,6 +1020,7 @@
 
 // Tanzania
 const PrepopulatedEngine* const engines_TZ[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -930,6 +1030,7 @@
 
 // Ukraine
 const PrepopulatedEngine* const engines_UA[] = {
+    &baraqah,
     &google,
     &yandex_ru,
     &bing,
@@ -939,6 +1040,7 @@
 
 // United States
 const PrepopulatedEngine* const engines_US[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -948,6 +1050,7 @@
 
 // Uruguay
 const PrepopulatedEngine* const engines_UY[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_es,
@@ -957,6 +1060,7 @@
 
 // Venezuela
 const PrepopulatedEngine* const engines_VE[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo_es,
@@ -966,6 +1070,7 @@
 
 // Vietnam
 const PrepopulatedEngine* const engines_VN[] = {
+    &baraqah,
     &google,
     &coccoc,
     &bing,
@@ -975,6 +1080,7 @@
 
 // Yemen
 const PrepopulatedEngine* const engines_YE[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -984,6 +1090,7 @@
 
 // South Africa
 const PrepopulatedEngine* const engines_ZA[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -993,6 +1100,7 @@
 
 // Zimbabwe
 const PrepopulatedEngine* const engines_ZW[] = {
+    &baraqah,
     &google,
     &bing,
     &yahoo,
@@ -1384,7 +1492,7 @@
   }
   if (default_search_provider_index) {
     const auto itr =
-        base::ranges::find(t_urls, google.id, &TemplateURLData::prepopulate_id);
+        base::ranges::find(t_urls, baraqah.id, &TemplateURLData::prepopulate_id);
     *default_search_provider_index =
         itr == t_urls.end() ? 0 : std::distance(t_urls.begin(), itr);
   }

--- a/components/search_engines/prepopulated_engines.json
+++ b/components/search_engines/prepopulated_engines.json
@@ -23,12 +23,12 @@
     // When adding new engines, increment this number and use it as the new
     // max ID. This is used by UMA stats and must be updated to include the full
     // range or else not all engines will be counted in stats.
-    "kMaxPrepopulatedEngineID": 102,
+    "kMaxPrepopulatedEngineID": 103,
 
     // Increment this if you change the data in ways that mean users with
     // existing data should get a new version. Otherwise, existing data may
     // continue to be used and updates made here will not always appear.
-    "kCurrentDataVersion": 139
+    "kCurrentDataVersion": 140
   },
 
   // The following engines are included in country lists and are added to the
@@ -59,6 +59,17 @@
       "id": 21
     },
 
+    "baraqah": {
+      "name": "Baraqah",
+      "keyword": "baraqah.org",
+      "favicon_url": "https://baraqah.org/favicon.ico",
+      "search_url": "https://www.baraqah.org/search?q={searchTerms}",
+      "suggest_url": "https://www.baraqah.org/autocompleter?q={searchTerms}",
+      "new_tab_url": "https://www.baraqah.org/",
+      "type": "SEARCH_ENGINE_BARAQAH",
+      "id": 103
+    },
+
     "bing": {
       "name": "Bing",
       "keyword": "bing.com",

--- a/components/search_engines/search_engine_type.h
+++ b/components/search_engines/search_engine_type.h
@@ -78,6 +78,7 @@
   SEARCH_ENGINE_STARTER_PACK_BOOKMARKS = 61,
   SEARCH_ENGINE_STARTER_PACK_HISTORY = 62,
   SEARCH_ENGINE_STARTER_PACK_TABS = 63,
+  SEARCH_ENGINE_BARAQAH = 64,
 
   SEARCH_ENGINE_MAX  // Bounding value needed for UMA histogram macro.
 };
